components/encryption/src/manager/mod.rs

```diff
@@ -149,6 +153,7 @@ impl Dicts {
     }
 
     fn save_key_dict(&self, master_key: &dyn Backend) -> Result<()> {
+        let _lk = self.key_dict_file_lock.lock().unwrap();
```

Comment:
When will it be rotated concurrently?

Reply:
After keys get bulk imported, `DataKeyImporter` will trigger a force rotate.